### ISSUE TITLE
Usage of field set to null as value source leads to somewhat cryptic error

### ISSUE THREAD

**Issue by estrizhok** [nunit/nunit#1121]
Consider the following code snippet.

```csharp
  public class TestClassFour
  {
    private static string sourceFour = "Rumpelstilzchen";
    public static string sourceFive = null;

    [Test]
    public void Four([ValueSource("sourceFour")] char source)
    {
      Assert.AreNotSame("z", source);
    }

    [Test]
    public void Five([ValueSource("sourceFive")] string source)
    {
      Assert.AreEqual("Argentina", source);
    }
  }
```
If one attempts to run tests from such class, he will encounter the following issues:

1. An error being reported is not user friendly enough.

> An exception was thrown while loading the test.
> System.NullReferenceException: Object reference not set to an instance of an object.
>   at NUnit.Framework.Internal.Builders.ParameterDataProvider.GetDataFor(IParameterInfo parameter)
>   at NUnit.Framework.CombiningStrategyAttribute.BuildFrom(IMethodInfo method, Test suite)
>   at NUnit.Framework.Internal.Builders.DefaultTestCaseBuilder.BuildFrom(IMethodInfo method, Test parentSuite)
>   at NUnit.Framework.Internal.Builders.NUnitTestFixtureBuilder.BuildTestCase(IMethodInfo method, TestSuite suite)
>   at NUnit.Framework.Internal.Builders.NUnitTestFixtureBuilder.AddTestCasesToFixture(TestFixture fixture)
>   at NUnit.Framework.Internal.Builders.NUnitTestFixtureBuilder.BuildFrom(ITypeInfo typeInfo)
>   at NUnit.Framework.Internal.Builders.DefaultSuiteBuilder.BuildFrom(ITypeInfo typeInfo)

2. Test cases from perfectly valid value source are not discovered and cannot be run until the error is fixed.

**Comment by rprouse** [comment]
Thanks for the report. It looks pretty serious, but should be fairly easy to fix.

**Comment by rprouse** [comment]
Fixed by PR #1184
